### PR TITLE
Add test for memory oversubscription.

### PR DESCRIPTION
**Pull request by shishir-a412ed**
@notnoop I am adding a test for #90 here.
Right now, it will fail. Once we merge #90, I ll rebase the PR, and the test should pass.

### CODE REVIEW

**Comment by notnoop** [comment]
Thanks. I have tested this and verified it works with PR #90.

We may need to tweak the scripts to enable memory oversubscription (it's opt-in) - with a step like:

```sh
curl -sSL  http://localhost:4646/v1/operator/scheduler/configuration \
    | jq '.SchedulerConfig + {MemoryOversubscriptionEnabled: true}' \
    | curl --request POST --data @- http://localhost:4646/v1/operator/scheduler/configuration
```

File: tests/001-test-redis.sh [hunk 1]
```diff
@@ -49,6 +49,19 @@ test_redis_nomad_job() {
        exit 1
     fi
 
+    echo "INFO: Check if memory and memory_max are set correctly in the cgroup filesystem."
+    task_name=$(sudo CONTAINERD_NAMESPACE=nomad ctr containers ls|awk 'NR!=1'|cut -d' ' -f1)
+    memory_soft_limit=$(sudo cat /sys/fs/cgroup/memory/nomad/$task_name/memory.soft_limit_in_bytes)
+    if [ $memory_soft_limit != "268435456" ]; then
+       echo "ERROR: memory should be 256 MB."
```
Comment (notnoop) [inline tests/001-test-redis.sh:56]: May I suggest using bash arithmetic here, hopefully to make understanding the test a bit easier:

```suggestion
    if [ $memory_soft_limit != "$(( 256 * 1024 * 1024 ))" ]; then
       echo "ERROR: memory should be 256 MB. Found ${memory_soft_limit}"
```

Comment (shishir-a412ed) [reply tests/001-test-redis.sh]: Fixed.

**Comment by shishir-a412ed** [comment]
@notnoop Please check now, if this looks good, I ll merge it.

**Review by notnoop — APPROVED** [review]
Looks good to me. `default_scheduler_config` is cleaner than the curl command indeed!

**Pull request merged**